test/integration/scheduler/scheduler_test.go

```diff
@@ -338,4 +368,168 @@ var _ = ginkgo.Describe("Scheduler", func() {
 			return createdJob2.Spec.Suspend
 		}, framework.Timeout, framework.Interval).Should(gomega.Equal(pointer.Bool(false)))
 	})
+
+	ginkgo.It("Should re-enqueue by the delete event of workload belonging to the same ClusterQueue", func() {
+		job1 := testing.MakeJob("on-demand-job1", ns.Name).Queue(prodBEQueue.Name).Request(corev1.ResourceCPU, "8").Obj()
+		gomega.Expect(k8sClient.Create(ctx, job1)).Should(gomega.Succeed())
+		workload1 := &kueue.QueuedWorkload{}
+		gomega.Eventually(func() bool {
+			lookupKey := types.NamespacedName{Name: job1.Name, Namespace: job1.Namespace}
+			err := k8sClient.Get(ctx, lookupKey, workload1)
+			return err == nil
+		}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+
+		createdJob1 := &batchv1.Job{}
+		gomega.Eventually(func() bool {
+			lookupKey := types.NamespacedName{Name: job1.Name, Namespace: job1.Namespace}
+			return k8sClient.Get(ctx, lookupKey, createdJob1) == nil && !*createdJob1.Spec.Suspend
+		}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
```

Comment:
```suggestion
		gomega.Eventually(func() *bool {
			lookupKey := types.NamespacedName{Name: job1.Name, Namespace: job1.Namespace}
			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob1)).Should(gomega.Succeed()) 
                         return createdJob1.Spec.Suspend
		}, framework.Timeout, framework.Interval).Should(gomega.Equal(pointer.Bool(false))
```

other places below too